src/Sylius/Bundle/CoreBundle/Listener/CatalogPromotionConfigurationRemovedListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Listener;
+
+use Sylius\Bundle\CoreBundle\Processor\CatalogPromotionClearerInterface;
+use Sylius\Bundle\CoreBundle\Processor\CatalogPromotionProcessorInterface;
+use Sylius\Component\Promotion\Event\CatalogPromotionConfigurationRemoved;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class CatalogPromotionConfigurationRemovedListener
+{
+    private CatalogPromotionClearerInterface $catalogPromotionClearer;
+
+    private CatalogPromotionProcessorInterface $catalogPromotionProcessor;
+
+    private RepositoryInterface $catalogPromotionRepository;
+
+    public function __construct(
+        CatalogPromotionClearerInterface $catalogPromotionClearer,
+        CatalogPromotionProcessorInterface $catalogPromotionProcessor,
+        RepositoryInterface $catalogPromotionRepository
+    ) {
+        $this->catalogPromotionClearer = $catalogPromotionClearer;
+        $this->catalogPromotionProcessor = $catalogPromotionProcessor;
+        $this->catalogPromotionRepository = $catalogPromotionRepository;
+    }
+
+    public function __invoke(CatalogPromotionConfigurationRemoved $event): void
+    {
+        $this->catalogPromotionClearer->clear();
+
+        foreach ($this->catalogPromotionRepository->findAll() as $catalogPromotion) {
+            $this->catalogPromotionProcessor->process($catalogPromotion);
+        }
```

Comment:
As this is the same as in other Listener maybe I will make CatalogPromotionReapplicator class so code won't be duplicated?